app/scripts/services/pods.js

```diff
@@ -40,6 +40,34 @@ angular.module("openshiftConsole")
         return debugPod;
       },
 
+      // Takes in a collection of pods and a collection of resources that
+      // control pods (replication controllers, replica sets, pet sets) and
+      // matches each pod to its owner.
+      //
+      // Returns: hash where the UID of the owner is the key and a hash of pods
+      //          is the value
+      groupByOwnerUID: function(pods, owners) {
+        var podsByOwnerUID = {};
+        // Look at each pod.
+        _.each(pods, function(pod) {
+          var found = false;
+          _.each(owners, function(owner) {
+            var selector = new LabelSelector(owner.spec.selector);
```

Comment:
instead of generating this selector for each owner for every single pod, can you do an initial pass on the owners and create a map of UID to LabelSelector object?